ERCS/erc-7646.md

```diff
@@ -0,0 +1,141 @@
+---
+eip: 7646
+title: Metadata Management for NFTs
+description: This ERC allows NFT owners and authorized addresses to update their token metadata.
+author: Alexey Kureev (@Kureev) <a.g.kureev@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/erc-7646-metadata-management-for-nfts/19027
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-03
+requires: 721, 1155
+---
+
+## Abstract
+
+The MetadataManager contract provides a mapping for storing metadata of NFTs. It also provides a mechanism for authorizing other addresses to update the metadata of a token. This contract can be used in conjunction with any NFT contract that implements the [ERC-721](./eip-721.md) or [ERC-1155](./eip-1155.md) standard.
+
+## Motivation
+
+Currently, the metadata of an NFT is immutable once it has been minted. This is problematic for use cases where the metadata of an NFT needs to be updated afterwards. For example, an NFT representing a digital artwork might need to have its metadata updated to reflect changes in the artwork's ownership or provenance.
+
+## Specification
+
+The MetadataManager contract provides the following functions:
+
+- `updateMetadata(address tokenAddress, uint256 tokenId, string memory newURI)`: Updates the metadata of a token. The sender must be the owner of the token or authorized to update the metadata.
+- `getMetadata(address tokenAddress, uint256 tokenId)`: Returns the metadata of a token. If the metadata is not stored, it will try to fetch it from the NFT contract.
+- `authorize(address tokenAddress, uint256 tokenId, address authorizedAddress)`: Authorizes an address to update the metadata of a token. Only the owner of the token can call this function.
+- `revokeAuthorization(address tokenAddress, uint256 tokenId, address authorizedAddress)`: Revokes the authorization of an address to update the metadata of a token. Only the owner of the token can call this function.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+## Rationale
+
+The rationale behind this proposal is to provide a decentralized method for the management of NFT metadata, allowing updates post-mint, as well as the ability for the owner to assign metadata editing permissions.
```

Comment:
Rationale is meant to explain design decisions you made within the proposal. This paragraph belongs in the Motivation section because it's justifying the proposal as a whole.